Automattic/newspack-blocks
# Donate block

### All Submissions:






### Changes proposed in this Pull Request:



This PR implements a Donate block. The block is intended to be used on a memberships landing page to capture donations, but it should be usable anywhere. 

- It can handle tiered and untiered donation settings.
- It's implemented using only HTML + CSS, so it should be AMP-compatible.
- The actual form handling when the 'Donate' button is clicked will be handled in a separate PR. 

Screenshots in details:

<details>

#### Tiered

![localhost_wp-admin_post php_post=100 action=edit](https://user-images.githubusercontent.com/7317227/62007456-72779000-b102-11e9-8aff-a875830f71cc.png)
![localhost_donations-block__newspack_donate=1 donation_value_once=20 88 donation_value_once_other=20 88 donation_frequency=month donation_value_month=30 donation_value_month_other=20 88 donation_value_year=360 don](https://user-images.githubusercontent.com/7317227/62007454-6ee40900-b102-11e9-9f8f-545995526254.png)
![localhost_donations-block__newspack_donate=1 donation_value_once=20 88 donation_value_once_other=20 88 donation_frequency=month donation_value_month=30 donation_value_month_other=20 88 donation_value_year=360 (1)](https://user-images.githubusercontent.com/7317227/62007455-70adcc80-b102-11e9-8fbe-6ab97e53f50c.png)

#### Untiered

![localhost_wp-admin_post php_post=100 action=edit (1)](https://user-images.githubusercontent.com/7317227/62007459-7dcabb80-b102-11e9-8e21-5186001681a3.png)
![localhost_donations-block__newspack_donate=1 donation_value_once=20 88 donation_value_once_other=20 88 donation_frequency=month donation_value_month=30 donation_value_month_other=20 88 donation_value_year=360 (2)](https://user-images.githubusercontent.com/7317227/62007457-74415380-b102-11e9-9535-593e0a2e13ae.png)


</details>

### How to test the changes in this Pull Request:
0. Build things: `npm run build:webpack`, etc.
1. If it's not merged into master, you'll need to check out the  https://github.com/Automattic/newspack-plugin/pull/199 branch.
2. Go to the donation settings, and set up donations.
3. Insert the Donate block on a post/page. Preview the block in the block editor and on the frontend.

### Other information:
